tests/data/modules_v2/refs/inmanta_plugins/refs/__init__.py

```diff
@@ -0,0 +1,85 @@
+import os
+import dataclasses
+
+from inmanta.references import reference, Reference, DataclassReference, RefValue
+from inmanta.plugins import plugin
+
+
+@reference("refs::Bool")
+class BoolReference(Reference[bool]):
+    """A reference to fetch environment variables"""
+
+    def __init__(self, name: str | Reference[str]) -> None:
+        """
+        :param name: The name of the environment variable.
+        """
+        super().__init__()
+        self.name = name
+
+    def resolve(self) -> bool:
+        """Resolve the reference"""
+        return os.getenv(self.resolve_other(self.name)) == "true"
+
+
+@reference("refs::String")
+class StringReference(Reference[str]):
+    """A reference to fetch environment variables"""
```

Comment:
Docstring is not accurate I think ? At this point in the review I don't fully understand resolve_other among other things